### PR TITLE
feat: hide owned selector picker if user doesn't own an entity

### PR DESCRIPTION
**Pull request by djamaile**
Signed-off-by: djamaile <rdjamaile@gmail.com>

## Hey, I just made a Pull Request!
This is a first attempt at #8198 

For example, if an user navigates to http://https://demo.backstage.io//catalog?filters%5Bkind%5D=user&filters%5Buser%5D=owned it will see that it owns zero users. It's good to not show that filter at all then. Currently this implementation for every type of Kind Entity.

#### :heavy_check_mark: Checklist

- [x ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ x] Screenshots attached (for UI changes)
- [ x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Screenshot 2021-11-23 at 22 50 04
Screenshot 2021-11-23 at 22 52 38
Screenshot 2021-11-23 at 22 52 55

### CODE REVIEW

**Review by adamdmharvey — COMMENTED** [review]
Very nice! 👏🏻 

I fed back in the related issue, but I'd offer we do NOT want to hide the owner filter for other kinds.  If I don't own any APIs, that's fine, but if I don't see the filter, as a user I may not even know to discover that it would be a possibility to do so. As a result, I may also therefore not attempt to TRY to own any as I may not see any value in doing so. "Oh, I really own Podcast API, but there's no point in putting my name as the real owner, it's fine if they just contact @djamaile instead."  Versus if I know it COULD be possible, I might try it so then I gain more value from the catalog.

For the Users and Groups, however, it's literally not possible, and so should be hidden.

Just one person's opinion :)

**Comment by djamaile** [comment]
@adamdmharvey I think that makes sense honestly. So, I was thinking for example if you don't own/starred an API  then those selectors will be grayed out and  'all' will be selected by default. What do you think about that? Also, 'owned' will be removed then only from groups and users. I do think that makes sense.

**Comment by adamdmharvey** [comment]
Yeah greyed out would make sense; sort of a visual indicator of why I can't use it!

**Comment by Rugvip** [comment]
@djamaile Yep! I think defaulting to "all" in case of no owned entities would be great but then keeping the other options around. I'm not sure we actually need to gray out the "owned" option, especially if it complicates the implementation. I feel it would be alright to be able to click "owned" and just see that the list is empty.

For users and groups, yeah we could probably remove the "owned" option altogether there.

File: plugins/catalog-react/src/components/UserListPicker/UserListPicker.tsx [hunk 1]
```diff
@@ -122,17 +128,30 @@ export const UserListPicker = ({
...
     .map(filterGroup => ({
       ...filterGroup,
       items: filterGroup.items.filter(
         ({ id }) => !availableFilters || availableFilters.includes(id),
       ),
     }))
     .filter(({ items }) => !!items.length);
+  const [filterGroups, setFilterGroups] =
+    useState<ButtonGroup[]>(initialFilterGroup);
+  const setDefaultFilterGroups = useCallback(() => {
```
Comment (Rugvip) [inline plugins/catalog-react/src/components/UserListPicker/UserListPicker.tsx:143]: Seems a little bit strange to have this defined outside as a callback tbh? Would it make sense to just inline this in the useEffect cleanup?

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

Few thoughts, and then we'll need to build green before we can :shipit:

**Comment by djamaile** [comment]
@Rugvip what do you think about it now? Tests is failing because of probably of some actions are happening in the `useEffect` and tests are not considering that. But you if you like this approach, next step would be fixing the tests.

**Comment by jhaals** [comment]
I'm not @Rugvip but it looks good to me :)

**Comment by Rugvip** [comment]
@djamaile I'd double check that the state + effect is needed at all. I'm feeling it should be possible to compute the `filterGroups` and `selectedUserFilter` that we use for the rendering just based on our current state and props. Always best to keep the amount of state and effects to a minimum.

**Comment by djamaile** [comment]
@Rugvip I removed the useEffect functionality, but one test is still failing because userFilter goes from `owned` to `all`. Any ideas on how to fix/change this?

File: backstage.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (benjdlambert) [inline backstage.json:1]: dont commit this

**Comment by freben** [comment]
Yeah not sure how to best fix that error - but it's still red!

**Comment by djamaile** [comment]
@freben yeah I think the error happens because `totalOwnedUserEntities ` is a value derived from `useContext`. I am not sure yet, I'll have a good look this weekend to see if I can find why this error happens.

**Review by Rugvip — APPROVED** [review]
Alright, thank you! 👍 😁 

Let's :shipit:

**Pull request merged**